pkg/resources/external_oauth_integration.go

```diff
@@ -41,548 +46,777 @@ var oauthExternalIntegrationSchema = map[string]*schema.Schema{
 		Optional:    true,
-		Description: "Specifies the list of roles that a client cannot set as the primary role. Do not include ACCOUNTADMIN, ORGADMIN or SECURITYADMIN as they are already implicitly enforced and will cause in-place updates.",
+		Description: "Specifies the list of roles that a client cannot set as the primary role. By default, this list includes the ACCOUNTADMIN, and SECURITYADMIN roles. To remove these privileged roles from the list, use the ALTER ACCOUNT command to set the EXTERNAL_OAUTH_ADD_PRIVILEGED_ROLES_TO_BLOCKED_LIST account parameter to FALSE.",
+		DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+			params := d.Get(ParametersAttributeName).([]any)
+			var found *sdk.Parameter
+			for _, v := range params {
+				param := v.(sdk.Parameter)
+				if param.Key == "EXTERNAL_OAUTH_ADD_PRIVILEGED_ROLES_TO_BLOCKED_LIST" {
+					found = &param
+				}
+			}
+			if found == nil || !helpers.StringToBool(found.Value) {
+				return false
+			}
+
+			return old == "ACCOUNTADMIN" || old == "SECURITYADMIN"
+		},
+		ConflictsWith: []string{"external_oauth_allowed_roles_list"},
 	},
-	"allowed_roles": {
-		Type:        schema.TypeSet,
-		Elem:        &schema.Schema{Type: schema.TypeString},
-		Optional:    true,
-		Description: "Specifies the list of roles that the client can set as the primary role.",
+	"external_oauth_allowed_roles_list": {
+		Type:          schema.TypeSet,
+		Elem:          &schema.Schema{Type: schema.TypeString},
+		Optional:      true,
+		Description:   "Specifies the list of roles that the client can set as the primary role.",
+		ConflictsWith: []string{"external_oauth_blocked_roles_list"},
 	},
-	"audience_urls": {
+	"external_oauth_audience_list": {
 		Type:        schema.TypeSet,
 		Elem:        &schema.Schema{Type: schema.TypeString},
 		Optional:    true,
 		Description: "Specifies additional values that can be used for the access token's audience validation on top of using the Customer's Snowflake Account URL ",
 	},
-	"any_role_mode": {
-		Type:        schema.TypeString,
-		Optional:    true,
-		Default:     string(snowflake.Disable),
-		Description: "Specifies whether the OAuth client or user can use a role that is not defined in the OAuth access token.",
-		ValidateFunc: validation.StringInSlice([]string{
-			string(snowflake.Disable),
-			string(snowflake.Enable),
-			string(snowflake.EnableForPrivilege),
-		}, true),
+	"external_oauth_any_role_mode": {
+		Type:         schema.TypeString,
+		Optional:     true,
+		Description:  fmt.Sprintf("Specifies whether the OAuth client or user can use a role that is not defined in the OAuth access token. Valid options are: %v", sdk.AsStringList(sdk.AllExternalOauthSecurityIntegrationAnyRoleModes)),
+		ValidateFunc: validation.StringInSlice(sdk.AsStringList(sdk.AllExternalOauthSecurityIntegrationAnyRoleModes), true),
 		DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
```

Comment:
Are those diff suppresses useful? Looking at the possible values they don't look that useful (same with other enum fields).

Reply:
Removed and added to migration guide.